src/wrapper_functions.cc

```diff
@@ -169,17 +170,16 @@ LibGitReference branch_lookup(git_repository* repo, const std::string& branch_na
     return { ref, git_reference_free };
 }
 
-// std::string branch_remote_name(git_repository* repo, const std::string& branch_name)
-// {
-//     git_buf buf;
-//     if (git_branch_remote_name(&buf, repo, branch_name.c_str()))
-//     {
-//         // gul14::cat("branch_remote_name: ", git_error_last()->message);
-//         return "";
-//     }
-//     auto ret = std::string{ buf.ptr };
-//     git_buf_dispose(&buf);
-//     return ret;
-// }
+std::string branch_remote_name(git_repository* repo, const std::string& branch_name)
+{
+    git_buf buf{ };
+    auto _ = gul14::finally([buf_addr = &buf]() { git_buf_dispose(buf_addr); });
+    auto result = git_branch_remote_name(&buf, repo, branch_name.c_str());
+    if (result) {
+        throw Error{ result, gul14::cat("branch_remote_name: ", git_error_last()->message) };
```

Comment:
I would find `error` more readable than `result` here, just because of the `if (error)`.

Reply:
Done.

While looking on this code I realized that `LibGitBuf` is probably useless (in `types.h`) - I tried to use it...